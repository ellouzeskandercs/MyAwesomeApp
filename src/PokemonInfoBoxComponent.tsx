import React from 'react';
import {View, StyleSheet, Text} from 'react-native';

const PokemonInfoBoxComponent = (props: {name: string}) => {
  return (
    <View style={styles.pokemonInfoBox}>
      <Text style={styles.pokemonBoxText}> {props.name} </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  pokemonInfoBox: {
    marginTop: 10,
    height: 40,
    width: '100%',
    borderRadius: 8,
    backgroundColor: 'powderblue',
    alignItems: 'center',
    justifyContent: 'center',
  },
  pokemonBoxText: {
    fontSize: 18,
  },
});

export {PokemonInfoBoxComponent};
