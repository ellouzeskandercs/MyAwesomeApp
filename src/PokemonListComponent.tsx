import React from 'react';
import {View, StyleSheet, Text, ScrollView} from 'react-native';

const PokemonInfoBox = (props: {name: string}) => {
  return (
    <View style={styles.pokemonInfoBox}>
      <Text style={styles.pokemonBoxText}> {props.name} </Text>
    </View>
  );
};

const PokemonListComponent = () => {
  return (
    <View style={styles.pokemonListContainer}>
      <Text style={styles.pokemonListHeader}>Welcome to the pokemon world</Text>
      <ScrollView>
        <PokemonInfoBox name="pokemon 1" />
        <PokemonInfoBox name="pokemon 2" />
        <PokemonInfoBox name="pokemon 3" />
        <PokemonInfoBox name="pokemon 4" />
        <PokemonInfoBox name="pokemon 5" />
        <PokemonInfoBox name="pokemon 6" />
        <PokemonInfoBox name="pokemon 7" />
        <PokemonInfoBox name="pokemon 8" />
        <PokemonInfoBox name="pokemon 9" />
        <PokemonInfoBox name="pokemon 10" />
        <PokemonInfoBox name="pokemon 11" />
        <PokemonInfoBox name="pokemon 12" />
        <PokemonInfoBox name="pokemon 13" />
        <PokemonInfoBox name="pokemon 14" />
        <PokemonInfoBox name="pokemon 15" />
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  pokemonInfoBox: {
    marginTop: 10,
    height: 40,
    width: '100%',
    borderRadius: 8,
    backgroundColor: 'powderblue',
    alignItems: 'center',
    justifyContent: 'center',
  },
  pokemonBoxText: {
    fontSize: 18,
  },
  pokemonListContainer: {
    paddingHorizontal: 10,
    paddingVertical: 20,
    flex: 1,
  },
  pokemonListHeader: {
    fontSize: 30,
    textAlign: 'center',
    fontWeight: 'bold',
  },
});

export default PokemonListComponent;
