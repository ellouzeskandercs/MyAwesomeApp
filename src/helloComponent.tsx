import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

const HelloComponent = () => {
  return (
    <View style={styles.helloContainer}>
      <Text style={styles.helloText}> Hello </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  helloContainer: {
    paddingHorizontal: 24,
    paddingVertical: 6,
    backgroundColor: 'red',
  },
  helloText: {
    color: 'green',
    fontSize: 24,
    fontWeight: '600',
  },
});

export default HelloComponent;
