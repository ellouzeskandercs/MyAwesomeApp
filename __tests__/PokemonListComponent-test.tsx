import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {PokemonListComponent} from '../src/PokemonListComponent';

it('renders correctly', () => {
  const component = renderer.create(<PokemonListComponent />);
  expect(component).toMatchSnapshot();
});
