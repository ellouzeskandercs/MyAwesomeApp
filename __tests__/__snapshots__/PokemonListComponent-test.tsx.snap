// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Welcome to the pokemon world
  </Text>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          bulbasaur
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          ivysaur
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          venusaur
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          charmander
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          charmeleon
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          charizard
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          squirtle
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          wartortle
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          blastoise
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          caterpie
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          metapod
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          butterfree
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          weedle
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          kakuna
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          beedrill
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          pidgey
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          pidgeotto
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          pidgeot
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          rattata
           
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "powderblue",
            "borderRadius": 8,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
          raticate
           
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
